yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -849,6 +852,27 @@ private[spark] class Client(
       }
     }
   }
+
+  private def cleanupStagingDir(): Unit = {
```

Comment:
This is mostly a copy and paste of the code on line 120 or so. Surely this can be refactored ?